anonlink/solving/_multiparty_solving_inner.cpp

```diff
@@ -0,0 +1,316 @@
+#include <algorithm>
+#include <cassert>
+#include <unordered_map>
+#include <vector>
+#include "_multiparty_solving_inner.h"
+
+namespace {
+
+struct RecordHasher {
+    std::hash<unsigned long long> hasher;
+    size_t operator()(const Record &record) const {
+        return hasher((unsigned long long)record.dset_i << 32 | record.rec_i);
```

Comment:
You still need to update this to use precise bit-width values (i.e. `std::uint64_t` _et al._).